rainier-core/src/main/scala/com/stripe/rainier/core/Categorical.scala

```diff
@@ -99,10 +99,27 @@ final case class Binomial(p: Real, k: Int) extends Distribution[Int] {
   val multi: Multinomial[Boolean] =
     Categorical.boolean(p).toMultinomial(k)
 
-  def generator: Generator[Int] =
-    multi.generator.map { m =>
+  def generator: Generator[Int] = {
+    val poissonGenerator = Poisson(p * k).generator.map { _.min(k) }
+    val normalGenerator = Normal(k * p, k * p * (1 - p)).generator.map {
+      _.toInt.max(0).min(k)
+    }
+    val binomialGenerator = multi.generator.map { m =>
       m.getOrElse(true, 0)
     }
+    Generator.from {
```

Comment:
weird formatting here, can you add more line breaks around the if/else clauses?

Reply:
I battled the auto-formatting and got it to look more reasonable.